test/framework/src/main/java/org/elasticsearch/index/mapper/FieldTypeTestCase.java

```diff
@@ -89,4 +97,56 @@ public static List<?> fetchStoredValue(MappedFieldType fieldType, List<Object> s
         fetcher.setNextReader(null);
         return fetcher.fetchValues(null, -1, new ArrayList<>());
     }
+
+    public void testFieldHasValue() {
+        MappedFieldType fieldType = getMappedFieldType();
```

Comment:
I know I suggested to put this in the base class, but this way we just run the same test, against a fake field type (besides when getMappedFieldType is overridden), and that seems wasteful. I was hoping that each subclass would provide its specific mapped field type impl and that the test would run against that. That would require extensive changes in that each subclass of FieldTypeTestCase should provide its own instance of mapped field type. We can do that as a follow-up, if we think it's important. 

For now, I do see why you introduced a different test class for these tests. That makes sense to me. Sorry about the back and forth.

Reply:
I see your point but I would keep tests as they are now. I see the wastefulness of running the same test against not overwritten fieldTypes but it sets a nice foundation for the follow-up work.

Reply:
as long as we do the follow-up work. I worry that many times these things are forgotten and stay in an inconsistent state for months/ years :)